rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/message/MessageFormLayout.java

```diff
@@ -5,43 +5,51 @@
 import android.os.Build;
 import android.support.annotation.Nullable;
 import android.text.Editable;
+import android.text.TextUtils;
 import android.text.TextWatcher;
 import android.util.AttributeSet;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.view.inputmethod.InputMethodManager;
 import android.widget.EditText;
 import android.widget.LinearLayout;
 import android.widget.TextView;
 
+import bolts.Continuation;
+import bolts.Task;
 import chat.rocket.android.widget.R;
+import chat.rocket.android.widget.internal.ExtraActionSelectionDialog;
 
-public class MessageComposer extends LinearLayout {
+public class MessageFormLayout extends LinearLayout {
+  public static final int EXTRA_ACTION_PICK_PHOTO = 1;
```

Comment:
Thank you for good advice.

I failed to consider that each server has different configurations about uploading feature.
I will fix...